app/models/piece.rb

```diff
@@ -59,6 +50,13 @@ def move_out_of_bounds?(to_row, to_col)
     to_row < 0 || to_row > 7 || to_col < 0 || to_col > 7
   end
 
+  def move_destination_same_side?(to_row, to_col)
+    if game.piece_at(to_row, to_col) && game.piece_at(to_row, to_col).is_black == is_black
```

Comment:
`piece_at` returns an array, that may or may not have a captured piece as an element. We need to extract the one un-captured element from the array.